### ISSUE TITLE
exclude master/client nodes from data requests

### ISSUE THREAD

**Issue by costin** [elastic/elasticsearch-hadoop#214]
Double check behaviour in cluster with master-only or client/tribe nodes in terms of writing/reading data.

**Comment by costin** [comment]
@jeffsteinmetz For some reason I've only found this comment now - apologies for the huge delay. THe latest Beta (4) has support for client only nodes - in other words, es-hadoop can be forced to connect to the cluster only through these nodes. Clearly it affects parallelism since the queries are distributed between these nodes (instead of going to the data nodes directly) but performance doesn't seem to be affected too much - depends on the volume really and how import locality is.
In other words, if you are doing HUGE bulk reads, you might find it slower, if not, you are unlikely to spot any difference.

Cheers,

**Comment by costin** [comment]
By the way, I've closed the issue since Beta4 was just released. Let me know if you have any issues/queries potentially through the mailing list or another issue.

Cheers!
